Source/Router.swift

```diff
@@ -210,14 +213,31 @@ public final class Router {
 
         let delayTime = dispatch_time(DISPATCH_TIME_NOW, Int64(latency * Double(NSEC_PER_SEC)))
         dispatch_after(delayTime, dispatch_get_main_queue()) {
-            didFinishLoading(server)
+            [weak self] in
+            // before reporting "finished", check if request has been canceled in the meantime
+            guard let strongSelf = self else {
+                return
+            }
+            if let serverIndex = strongSelf.canceledRequests.indexOf(server) {
+                // remove server from the list of "canceled requests" and DO NOT send notification(s)
+                strongSelf.canceledRequests.removeAtIndex(serverIndex)
+            }
+            else {
+                didFinishLoading(server)
+            }
+        }
+    }
+
+    func stopLoading(server: NSURLProtocol) {
+        if canceledRequests.contains(server) == false {
```

Comment:
Server keeps track of registered Router objects, so a flag on Routers would be a possible solution I guess. No need to store Server references from Routers as I commented before (which actually doesn't make that much sense, we will never have multiple Server instances).

Other than that, unit testing this new behaviour should also show how valid the approach is.